panther_analysis_tool/validation.py

```diff
@@ -101,15 +103,9 @@ def validate_packs(analysis_specs: ClassifiedAnalysisContainer) -> List[Any]:
     id_to_detection = {}
     for item in analysis_specs.items():
         analysis_spec = item.analysis_spec
-        analysis_id = (
-            analysis_spec.get("PolicyID")
-            or analysis_spec.get("RuleID")
-            or analysis_spec.get("DataModelID")
-            or analysis_spec.get("GlobalID")
-            or analysis_spec.get("PackID")
-            or analysis_spec.get("QueryName")
-            or analysis_spec["LookupName"]
-        )
+        analysis_id = lookup_analysis_id(analysis_spec)
+        if analysis_id == UNKNOWN_ANALYSIS_ID:
+            raise UnknownAnalysisIdError(f"Unknown analysis id from analysis item {analysis_spec}")
```

Comment:
does this need to be caught anywhere?

Reply:
I don't expect this to happen. But in the off chance it does, it will be caught at the highest level and be printed as an error